sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsAsyncClient.java

```diff
@@ -0,0 +1,228 @@
+
+/**
+ * The asynchronous client for querying Azure Monitor logs.
+ */
+@ServiceClient(builder = LogsClientBuilder.class, isAsync = true)
+public final class LogsAsyncClient {
+
+    private final AzureLogAnalyticsImpl innerClient;
+
+    /**
+     * Constructor that has the inner generated client to make the service call.
+     * @param innerClient The inner generated client.
+     */
+    LogsAsyncClient(AzureLogAnalyticsImpl innerClient) {
+        this.innerClient = innerClient;
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given query in the specified workspaceId.
+     * @param workspaceId The workspaceId where the query should be executed.
+     * @param query The Kusto query to fetch the logs.
+     * @param timeSpan The time period for which the logs should be looked up.
+     * @return The logs matching the query.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<LogsQueryResult> queryLogs(String workspaceId, String query, QueryTimeSpan timeSpan) {
+        return queryLogsWithResponse(new LogsQueryOptions(workspaceId, query, timeSpan))
+                .map(Response::getValue);
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given query in the specified workspaceId.
+     * @param options The query options.
+     * @return The logs matching the query.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<LogsQueryResult>> queryLogsWithResponse(LogsQueryOptions options) {
+        return withContext(context -> queryLogsWithResponse(options, context));
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given batch of queries in the specified workspaceId.
+     * @param workspaceId The workspaceId where the batch of queries should be executed.
+     * @param queries A batch of Kusto queries.
+     * @param timeSpan The time period for which the logs should be looked up.
+     * @return A collection of query results corresponding to the input batch of queries.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<LogsQueryBatchResultCollection> queryLogsBatch(String workspaceId, List<String> queries,
+                                                               QueryTimeSpan timeSpan) {
+        LogsQueryBatch logsQueryBatch = new LogsQueryBatch();
+        queries.forEach(query -> logsQueryBatch.addQuery(workspaceId, query, timeSpan));
+        return queryLogsBatchWithResponse(logsQueryBatch).map(Response::getValue);
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given batch of queries.
+     * @param logsQueryBatch {@link LogsQueryBatch} containing a batch of queries.
+     * @return A collection of query results corresponding to the input batch of queries.@return
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<LogsQueryBatchResultCollection>> queryLogsBatchWithResponse(LogsQueryBatch logsQueryBatch) {
```

Comment:
I am surprised batch and single calls here are not polled calls. How long can they take to return results? 

Do we expect the consumers of it to be tools like Kusto explorers or direct customers who want to query?

Reply:
Yeah, the service doesn't support long running operations. The results can take up to 10 minutes for a complex query and the user can set server timeout to indicate that they are willing to wait for a specific duration.